<launch>
    <arg name="vrpn" default="false" doc="Using vrpn system?"/>
    <arg name="type" default="3D" doc="2D or 3D radar data"/>
    <arg name="odr_flag" default="false" doc="true:=publish ODR estimate, false:=publish mlesac estimate"/>
    <arg name="mmwave_topic" default="/mmWaveDataHdl/RScan" doc=""/>

    <arg name="azimuth_thres" default="80" doc="Azimuth threshold value"/>
    <arg name="elevation_thres" default="80" doc="Elevation threshold value"/>
    <arg name="range_thres" default="0.30" doc="Range threshold value"/>
    <arg name="intensity_thres" default="5" doc="Intensity threshold value"/>

  <!-- launch velocity_estimator_node -->
    <node pkg="goggles" type="velocity_estimator_node_v2.py" name="goggles_node" output="screen">
        <param name="type" value="$(arg type)"/>
        <param name="odr_flag" value="$(arg odr_flag)"/>
        <param name="mmwave_topic" value="$(arg mmwave_topic)"/>

        <param name="azimuth_thres" value="$(arg azimuth_thres)"/>
        <param name="elevation_thres" value="$(arg elevation_thres)"/>
        <param name="range_thres" value="$(arg range_thres)"/>
        <param name="intensity_thres" value="$(arg intensity_thres)"/>
    </node>

    <!-- launch inertial2BodyFrame Velocity transform node -->

    <node if="$(arg vrpn)" pkg="static_tfs" type="vrpn2BodyFrameVelocity.py" name="vrpn2BodyFrameVelocity_node" output="screen"/>

</launch>
